File: 卡码/代码随想录/1/06-二叉树/Python/03-二叉树修改与构造/LeetCode_0654.py
#654. 最大二叉树
from typing import *
from ..TreeNode import *

class Solution:
  def traversal(self,nums:List[int],left:int,right:int)->Optional[TreeNode]:
    if left>=right:
      return None
    maxIndex:int=left
    for i in range(left,right):
      if nums[i]>nums[maxIndex]:
        maxIndex=i
    root:TreeNode=TreeNode(nums[maxIndex])
    root.left=self.traversal(nums,left,maxIndex)
    root.right=self.traversal(nums,maxIndex+1,right)
    return root
  def constructMaximumBinaryTree(self, nums: List[int]) -> Optional[TreeNode]:
    return self.traversal(nums,0,len(nums))

File: 卡码/代码随想录/1/06-二叉树/Go/03-二叉树修改与构造/LeetCode_0654.go
//654. 最大二叉树
package main

import (
	. "binaryTree/types"
)

func traversal(nums []int,left,right int) *TreeNode{
	if left>=right {
		return nil
	}
	maxIndex:=left
	for i:=left;i<right;i++ {
		if nums[i]>nums[maxIndex] {
			maxIndex=i
		}
	}
	root:=&TreeNode{nums[maxIndex],nil,nil}
	root.Left=traversal(nums,left,maxIndex)
	root.Right=traversal(nums,maxIndex+1,right)
	return root
}
func constructMaximumBinaryTree(nums []int) *TreeNode {
	return traversal(nums,0,len(nums))
}
